File: lismove-android/app/src/main/java/it/lismove/app/android/awards/data/AwardAchievement.kt
package it.lismove.app.android.awards.data

data class AwardAchievement(
    val achievement: Long,
    val description: String,
    val id: Long,
    val imageUrl: String?,
    val name: String = "",
    val type: Int,
    val value: Double
){
    fun getTypeLabel(): String{
        return when(type){
            TYPE_MONEY -> "euro"
            TYPE_POINTS -> "punti"
            else -> throw Exception("Type differs from expected")
        }
    }
    companion object{
        const val TYPE_MONEY = 0
        const val TYPE_POINTS = 1
    }
}

File: lismove-iOS/LisMove/Modules/Shared/Model/Award/AwardAchievement.swift
//
//  Award.swift
//  LisMove
//
//  Created by Francesco Paolo Dellaquila on 06/09/21.
//

import Foundation

public class AwardAchievement: Codable {
    var achievement: Int?
    var description: String?
    var id: Int?
    var imageURL, name: String?
    var value: Int?
    var type: AwardAchievementType?
    
    public enum AwardAchievementType: Int, Codable{
        case TYPE_MONEY = 0
        case TYPE_POINTS = 1
    }
    
    enum CodingKeys: String, CodingKey{
        case achievement = "achievement"
        case description = "description"
        case id = "id"
        case imageURL = "imageUrl"
        case name = "name"
        case type = "type"
        case value = "value"
    }
    
    

    public init(description: String?, id: Int?, imageURL: String?, name: String?, type: AwardAchievementType?, value: Int?) {
        self.description = description
        self.id = id
        self.imageURL = imageURL
        self.name = name
        self.type = type
        self.value = value
    }
    
    func getTypeLabel() -> String{
        switch type {
        case .TYPE_MONEY: return "euro"
        case .TYPE_POINTS: return "punti"
        default: return ""
        }
    }
}
